src/main/psxsdk/libspu/s_n2p.c

```diff
@@ -1,5 +1,42 @@
 #include "common.h"
 
-INCLUDE_ASM("main/nonmatchings/psxsdk/libspu/s_n2p", _spu_note2pitch);
+extern s16 D_80033860[];
+
+s32 _spu_note2pitch(s32 center_note, u16 center_note_frac, s32 offset_note,
+                    u16 offset_note_frac) {
+    s16 octave;
+    s32 temp_v0_3;
+    s16 var_a1;
+    s32 var_v0;
+    s16 var_v1_2;
+    u16 sample_rate;
+    s32 pos;
+    var_v0 = offset_note_frac + center_note_frac;
+    if (var_v0 < 0) {
+        var_v0 += 7;
+    }
+    var_a1 = var_v0 >> 3;
+    var_v1_2 = 0;
+    if (var_a1 >= 16) {
```

Comment:
You can convert the if/do/while into just a while loop to save lines of code and nesting: https://decomp.me/scratch/mr9Ya